mne_bids/mne_bids.py

```diff
@@ -158,6 +158,68 @@ def _events_tsv(events, raw, fname, trial_type, verbose):
     return fname
 
 
+def _participants_tsv(raw, group, fname, verbose):
+    """Create a participants.tsv file and save it.
+
+    This will append any new participant data to the current list if it
+    exists
+
+    Parameters
+    ----------
+    raw : instance of Raw
+        The data as MNE-Python Raw object.
+    group : str
+        Name of group participant belongs to.
+    fname : str
+        Filename to save the participants.tsv to.
+    verbose : bool
+        Set verbose output to true or false.
+
+    """
+    subject_info = raw.info['subject_info']
+    if subject_info is not None:
+        genders = {0: 'U', 1: 'M', 2:'F'}
+        sex = genders[subject_info.get('sex', 0)]
+        # this will have issues if the his_id doesn't actually match the
+        # subject id provided...
+        subject_id = 'sub-' + subject_info['his_id']
+        try:
+            age = subject_info['birthday']
+            bday = datetime(age[0], age[1], age[2])
+            meas_date = raw.info['meas_date']
+            if isinstance(meas_date, (np.ndarray, list)):
+                meas_date = meas_date[0]
+            meas_date = datetime.fromtimestamp(meas_date)
+            subject_age = age_on_date(bday, meas_date)
+        except KeyError:
+            subject_age = "n/a"
+        if os.path.exists(fname):
+            df = pd.read_csv(fname, sep='\t')
+            df = df.append(pd.DataFrame(data={'participant_id': [subject_id],
+                                              'age': [subject_age],
+                                              'sex': [sex],
+                                              'group': [group]},
+                                        columns=['participant_id', 'age',
+                                                 'sex', 'group']))
+            df = df.drop_duplicates()
+            df = df.sort_values(by='participant_id')
+        else:
+            df = pd.DataFrame(data={'participant_id': [subject_id],
```

Comment:
maybe define the `data` outside the `if-else` so we avoid breaking it into 6 lines ...